plugins/module_utils/redfish_utils.py

```diff
@@ -65,6 +69,16 @@ def get_request(self, uri):
         req_headers = dict(GET_HEADERS)
         username, password, basic_auth = self._auth_params(req_headers)
         try:
+            if "UserName" in self.creds:
+                username = self.creds["UserName"]
+                password = self.creds["Password"]
+            else:
+                username = self.creds["user"]
+                password = self.creds["pswd"]
+            if self.creds.get("Session") is not None:
+                req_headers["X-Auth-Token"] = self.creds.get("Session")
+                username = None
+                password = None
```

Comment:
I think this whole block should happen in `self._auth_params`. On the other hand, I don't think this is really needed, since `self._auth_params` already supports what you need? You just have to pass in the correct information in `self.creds`.

Reply:
@felixfontein @mraineri  -  We are not aware of this function self._auth_params as it is not available in the code when Ansible is installed freshly. We feel that we could have been informed about this at the initial stage of the PR. We would also like to know why this implementation is not available when a user installs Ansible. 
If there is a similar implementation already available, what would be the fate of this PR? Please clarify.